jenkins/spark-tests.sh

```diff
@@ -209,17 +211,19 @@ run_iceberg_tests() {
   ICEBERG_VERSION=${ICEBERG_VERSION:-0.13.2}
   # get the major/minor version of Spark
   ICEBERG_SPARK_VER=$(echo $SPARK_VER | cut -d. -f1,2)
+  IS_SPARK_33_OR_LATER=0
+  [[ "$(printf '%s\n' "3.3" "$ICEBERG_SPARK_VER" | sort -V | head -n1)" = "3.3" ]] && IS_SPARK_33_OR_LATER=1
 
-  # Iceberg does not support Spark 3.3+ yet
-  if [[ "$ICEBERG_SPARK_VER" < "3.3" ]]; then
-    PYSP_TEST_spark_jars_packages=org.apache.iceberg:iceberg-spark-runtime-${ICEBERG_SPARK_VER}_2.12:${ICEBERG_VERSION} \
+  # RAPIDS-iceberg does not support Spark 3.3+ yet
+  if [[ "$IS_SPARK_33_OR_LATER" -eq "1" ]]; then
```

Comment:
Nit: We should be consistent, using `=` above with double brackets, yet `-eq` here.
```suggestion
  if [[ "$IS_SPARK_33_OR_LATER" = "1" ]]; then
```